addons/account/models/chart_template.py

```diff
@@ -136,6 +151,78 @@ class AccountChartTemplate(models.Model):
         help="Account that will be set on lines created in cash basis journal entry and used to keep track of the "
              "tax base amount.")
 
+    @api.constrains('currency_id', 'parent_id')
+    def _constraints_required_parent(self):
+        for record in self:
+            if not record.currency_id and not record.parent_id.currency_id:
+                raise ValidationError(_("No currency set on %s nor its parent"))
+            if not record.bank_account_code_prefix and not record.parent_id.bank_account_code_prefix:
+                raise ValidationError(_("bank account code prefix set on %s nor its parent"))
+            if not record.cash_account_code_prefix and not record.parent_id.cash_account_code_prefix:
+                raise ValidationError(_("cash account code prefix set on %s nor its parent"))
+            if not record.transfer_account_code_prefix and not record.parent_id.transfer_account_code_prefix:
+                raise ValidationError(_("transfer account code prefix set on %s nor its parent"))
+
+    @api.depends('code_digits', 'parent_id')
+    def _compute_code_digits(self):
+        for record in self:
+            record.code_digits = record.code_digits or record.parent_id.code_digits or 6
+
+    @api.depends('code_digits', 'parent_id')
+    def _inverse_compute_code_digits(self):
+        for record in self:
+            record.code_digits = record.code_digits
```

Comment:
What? This seems useless; also there is no depends for inverse fields, it doesn't really make sense to have one.
If you have a second field like suggested, then you could write on the same record's stored field